src/services/geetest.ts

```diff
@@ -46,17 +52,28 @@ const GeeTest = (config): GeeTestType => {
     return JSON.parse(result.data)
   }
 
-  const validate = async (challenge, validate, seccode) => {
-    const gtLib = new GeetestLib(config.id, config.key)
-    const bypasscache = await getBypassStatus() // not a cache
-    let result
-    const params = []
-    if (bypasscache === "success") {
-      result = await gtLib.successValidate(challenge, validate, seccode, params)
-    } else {
-      result = gtLib.failValidate(challenge, validate, seccode)
+  const validate = async (
+    challenge,
+    validate,
+    seccode,
+  ): Promise<boolean | GeetestError> => {
+    try {
+      const gtLib = new GeetestLib(config.id, config.key)
+      const bypasscache = await getBypassStatus() // not a cache
+      let result
+      const params = []
+      if (bypasscache === "success") {
+        result = await gtLib.successValidate(challenge, validate, seccode, params)
+      } else {
+        result = gtLib.failValidate(challenge, validate, seccode)
+      }
+      if (result.status !== 1) {
+        return new GeetestUserFailToPassError()
+      }
+      return true
+    } catch (err) {
+      return new UnknownGeetestError()
```

Comment:
Small nit, we try to do this where possible to get visibility on the unknown error later on.
```suggestion
      return new UnknownGeetestError(err)
```